### PR TITLE
DxController Refactor

### PR DESCRIPTION
**Pull request by mprasanjith**

### CODE REVIEW

File: contracts/dxdao/DxController.sol [hunk 1]
```diff
@@ -1,7 +1,216 @@
...
+        bool isRegistered;
+        bool canManageSchemes;
+        bool canMakeAvatarCalls;
+    }
+
+    address[] schemesAddresses;
+    mapping(address => Scheme) public schemes;
+    uint256 schemesWithManageSchemesPermission;
+
+    DxAvatar public avatar;
```
Comment (AugustoL) [inline contracts/dxdao/DxController.sol:24]: Lets remove avatar, is not being used in the contract, it is required as parameter in some cases but with no need. Remove this, the onlyValidAvatar modifier and parameter in the register/unregister functions

File: contracts/dxdao/DxController.sol [hunk 2]
```diff
@@ -1,7 +1,216 @@
...
+        schemes[_scheme].isRegistered = false;
+        return true;
+    }
+
+    /**
+     * @dev unregister the caller's scheme
+     * @return bool which represents a
+      success
+     */
+    function unregisterSelf(address _avatar) external onlyRegisteredScheme isAvatarValid(_avatar) returns (bool) {
```
Comment (AugustoL) [inline contracts/dxdao/DxController.sol:138]: Remove this function please, we will use unregisterScheme instead

File: contracts/dxdao/DxController.sol [hunk 3]
```diff
@@ -1,7 +1,216 @@
...
+ */
+contract DxController is Initializable {
+    struct Scheme {
+        bytes32 paramsHash; // a hash "configuration" of the scheme
+        bool isRegistered;
+        bool canManageSchemes;
+        bool canMakeAvatarCalls;
+    }
+
+    address[] schemesAddresses;
```
Comment (AugustoL) [inline contracts/dxdao/DxController.sol:20]: This can be public

File: contracts/dxdao/DxController.sol [hunk 4]
```diff
@@ -1,7 +1,216 @@
...
+    mapping(address => Scheme) public schemes;
+    uint256 schemesWithManageSchemesPermission;
```
Comment (AugustoL) [inline contracts/dxdao/DxController.sol:22]: This can be public

**Review by AugustoL — CHANGES_REQUESTED** [review]

File: contracts/dxdao/DxController.sol (hunk 1 above)
Comment (mprasanjith) [reply contracts/dxdao/DxController.sol]: `avatar` is being used in the `avatarCall` function, so not sure if I should still remove this.

**Review by AugustoL — CHANGES_REQUESTED** [review]
Remove the changes done in DaoCreator please :grimacing:

**Review by AugustoL — APPROVED** [review]

**Pull request merged**